octopoes/octopoes/events/events.py

```diff
@@ -33,7 +33,9 @@ class OOIDBEvent(DBEvent):
 
     @property
     def primary_key(self) -> str:
-        return self.new_data.primary_key if self.new_data else self.old_data.primary_key
+        # There doesn't seem to be an easy way to tell mypy that if new_data is
+        # None then old_data is never None.
+        return self.new_data.primary_key if self.new_data else self.old_data.primary_key  # type: ignore[union-attr]
```

Comment:
Definining the event types better and not have an old_data and new_data which are `None` depending on the operation_type would fix this. I think we also want to change the way this works, so I decided not to refactor everything and just ignore the type errors in a lot of places.